sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -1350,7 +1352,6 @@ pi_result piDevicesGet(pi_platform Platform, pi_device_type DeviceType,
 
   PI_ASSERT(Platform, PI_INVALID_PLATFORM);
 
-  std::lock_guard<std::mutex> Lock(Platform->PiDevicesCacheMutex);
```

Comment:
Don't we need PiDevicesCacheMutex lock at line 1362?

Reply:
I don't think we need to lock the map for read-only access

Reply:
It is going to be a problem if there is another thread writing to the cache while this thread is reading the cache at the same time.

Reply:
No one should be reading the cache until it is populated. This is guaranteed by calling to populateDeviceCacheIfNeeded before attempting the read

Reply:
Hmm.. I thought piextDeviceCreateWithNativeHandle() can call populateDeviceCacheIfNeeded() in one thread.
And another thread can call piDevicesGet() at the same time.
Is this impossible?

Reply:
It possible and would be OK. Both piextDeviceCreateWithNativeHandle and piDevicesGet first call to populateDeviceCacheIfNeeded, which does the lock. So only one thread will populate the cache and others will wait. After that all accesses to the cache are read-only

Reply:
I see. It makes sense.
piDevicesGet() will wait until the other writer is done.
Thanks for your clarification.